File: client/src/services/resourceService.ts
import api from './api';

export interface Resource {
  id: string;
  name: string;
  type: string;
  category?: string;
  description?: string;
  availability: string;
  capacity: number;
  currentUtilization: number;
  costPerHour: number;
  location?: string;
  specifications?: any;
  healthCenterId?: string;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface ResourceCreate {
  name: string;
  type: string;
  category?: string;
  description?: string;
  availability?: string;
  capacity?: number;
  currentUtilization?: number;
  costPerHour?: number;
  location?: string;
  specifications?: any;
  healthCenterId?: string;
}

export interface ResourceUpdate {
  name?: string;
  type?: string;
  category?: string;
  description?: string;
  availability?: string;
  capacity?: number;
  currentUtilization?: number;
  costPerHour?: number;
  location?: string;
  specifications?: any;
  healthCenterId?: string;
}

export const resourceService = {
  // Obtener todos los recursos
  getAll: async (params?: { skip?: number; limit?: number; resource_type?: string; center_id?: string }) => {
    const response = await api.get('/api/v1/config/resources', { params });
    return response.data;
  },

  // Obtener recurso por ID
  getById: async (id: string) => {
    const response = await api.get(`/api/v1/config/resources/${id}`);
    return response.data;
  },

  // Crear nuevo recurso
  create: async (resource: ResourceCreate) => {
    const response = await api.post('/api/v1/config/resources', resource);
    return response.data;
  },

  // Actualizar recurso
  update: async (id: string, resource: ResourceUpdate) => {
    const response = await api.put(`/api/v1/config/resources/${id}`, resource);
    return response.data;
  },

  // Eliminar recurso
  delete: async (id: string) => {
    const response = await api.delete(`/api/v1/config/resources/${id}`);
    return response.data;
  },

  // Obtener recursos por centro
  getByCenter: async (centerId: string) => {
    const response = await api.get(`/api/v1/config/resources/center/${centerId}`);
    return response.data;
  },

  // Obtener recursos por tipo
  getByType: async (type: string) => {
    const response = await api.get(`/api/v1/config/resources/type/${type}`);
    return response.data;
  },
};

File: backend/app/services/resource_service.py
"""
Servicio para gestión de recursos
"""

from sqlalchemy.orm import Session
from typing import List, Optional
import uuid
from datetime import datetime

from app.schemas.resource import ResourceCreate, ResourceUpdate, ResourceResponse

class ResourceService:
    """Servicio para operaciones con recursos"""
    
    @staticmethod
    async def get_all(
        db: Session,
        skip: int = 0,
        limit: int = 100,
        resource_type: Optional[str] = None,
        center_id: Optional[str] = None
    ) -> List[ResourceResponse]:
        """Obtener todos los recursos"""
        # Placeholder - implementar con base de datos real
        return []
    
    @staticmethod
    async def create(db: Session, resource: ResourceCreate) -> ResourceResponse:
        """Crear nuevo recurso"""
        # Placeholder - implementar con base de datos real
        resource_id = str(uuid.uuid4())
        return ResourceResponse(
            id=resource_id,
            name=resource.name,
            type=resource.type,
            isActive=True,
            createdAt=datetime.utcnow(),
            updatedAt=datetime.utcnow()
        )





